deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sResourceDispatcher.java

```diff
@@ -101,20 +103,37 @@ public void apply(List<Manifest> manifests) throws HyscaleException {
             throw new HyscaleException(DeployerErrorCodes.MANIFEST_REQUIRED);
         }
         createNamespaceIfNotExists();
+        try {
+            List<Manifest> manifestWithHandlers = new ArrayList<>();
+            List<Manifest> manifestWithoutHandlers = new ArrayList<>();
+            for (Manifest manifest : manifests) {
+                if ((ResourceHandlers.getHandlerOf(KubernetesResourceUtil.
+                        getKubernetesResource(manifest, namespace).getKind()) != null) ?
+                        manifestWithHandlers.add(manifest) : manifestWithoutHandlers.add(manifest)) ;
+            }
+            List<KubernetesResource> k8sResources = getSortedResources(manifestWithHandlers);
+            List<String> appliedKinds = buildAppliedKindsAnnotation(getCustomObjects(manifests));
+            MultiValueMap<String, CustomObject> kindVsCustomObjects = getCustomObjects(manifestWithoutHandlers);
+            applyK8sResources(k8sResources, appliedKinds);
+            applyCustomResources(kindVsCustomObjects);
+            logger.info("Successfully applied all Manifests");
+        } catch (Exception e) {
+            HyscaleException ex = new HyscaleException(e, DeployerErrorCodes.FAILED_TO_APPLY_MANIFEST);
+            logger.error("Error while applying manifests to kubernetes", ex);
+            throw ex;
+        }
+    }
```

Comment:
Instead of reading data from manifest multiple times, we can read them once and use it again, and change the other functions accordingly